### PR TITLE
refactoring the selftest workflow

### PR DESCRIPTION
**Pull request by jspaaks**
##Description

This PR adds self tests for the various usage modes of `action.yml`, e.g.

- run on checked out version of your own repo
- with `repositories`
- with `curations` from a url
- with `curations` from a local file
- with `rules` from a url
- with `rules` from a local file
- with `classifications` from a url
- with `classifications` from a local file

TODO: https://github.com/tortellini-tools/action/pull/109#discussion_r644703341

Refs: #110

## Checklist

This section should be filled in by the creator of this pull request to make sure the pull request is ready to review.

- [x] This pull request has a descriptive title
- [x] Code is written according to the code quality guidelines
- [x] Documentation if applicable is available
- [x] Tests run successfully
- [x] All checks below this pull request were successful

## Instructions to review the pull request

Github Action `testing/*` should be green

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -23,7 +23,30 @@ jobs:
...
+        - name: Clone this repository
+          uses: actions/checkout@v2
+
+
+        - name: test without any input arguments
+          uses: ./
+        - name: Check the resulting directory tree
+          run: tree -a -L 5 -I '.git' .
+
+        - name: thename
```
Comment (sverhoeven) [inline .github/workflows/testing.yml]: Can we make 2 separate jobs so tests dont interfere with each other.
```suggestion
  test-action-multiple: 
    name: Test action with multiple repositories
    runs-on: ubuntu-latest
    steps:
        - name: thename
```

Comment (jspaaks) [reply .github/workflows/testing.yml]: yep still working on it

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -11,19 +11,145 @@ jobs:
...
+      runs-on: ubuntu-latest
+      steps:
+        - name: Clone this repository
+          uses: actions/checkout@v2
+        - name: Test without any input arguments
+          uses: ./
+        - name: Check the resulting directory tree
+          run: tree -a -L 5 -I '.git' .
+        - name: Verify that we have the analyze result
+          run: ls -l out/analyzer-result.yml
```
Comment (fdiblen) [inline .github/workflows/testing.yml]: Can you also run `ls` to check
analyzer-result.yml
evaluation-result.yml
gl-license-scanning-report.json

Comment (fdiblen) [reply .github/workflows/testing.yml]: also use `.tortellini/out/analyzer-result.yml`

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -11,19 +11,145 @@ jobs:
...
+      - run: npm run build
+      - run: npm run package
+      - run: npm run test
+
+
+  no-args:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Clone this repository
+          uses: actions/checkout@v2
```
Comment (fdiblen) [inline .github/workflows/testing.yml:24]: run `pwd` to see work directory.

File: .github/workflows/usage.yml [hunk 4]
```diff
@@ -20,7 +20,7 @@ jobs:
       - run: |
           echo 'https://github.com/tortellini-tools/action' > urls.txt
           echo 'https://github.com/fair-software/howfairis' >> urls.txt
-      - uses: tortellini-tools/action@81-config-options-with-additions
+      - uses: tortellini-tools/action@main
         with:
           repositories: 'urls.txt'
```
Comment (fdiblen) [inline .github/workflows/usage.yml]:
```suggestion
          repositories: urls.txt
```

**Review by fdiblen — APPROVED** [review]
Nice work @jspaaks!!! :sake:

**Comment by jspaaks** [comment]
Thanks for reviewing :milk_glass: :cookie:

**Pull request merged**